### DISCUSSION TITLE
How do I output results for each category during validation?

### DISCUSSION THREAD

**Discussion by d710055071** [open-mmlab/mmclassification#1422] [Q&A]
How do I output results for each category during validation?
for example
                  precision    recall  f1-score   support

           5     0.9272    0.9955    0.9601      1331
          10     0.1672    0.7183    0.2713        71
          15     0.2586    0.8824    0.4000        17
          20     0.8351    0.9766    0.9003      1110
          30     0.9873    0.9455    0.9660      6886
          40     0.9690    0.9767    0.9729      7266
          50     0.9948    0.7495    0.8549      5074
          60     0.8506    0.9758    0.9089      5380
          70     0.9986    0.9257    0.9608      1575
          80     0.9233    0.9666    0.9445      3738
         100     0.9657    0.9876    0.9765      1369
         120     0.2679    0.9375    0.4167        16

    accuracy                         0.9332     33833
   macro avg     0.7621    0.9198    0.7944     33833
weighted avg     0.9456    0.9332    0.9351     33833

**Comment by Ezra-Yu** [comment]
in config file:

```
val_evaluator = dict(type='ConfusionMatrix')
```

or 

```
val_evaluator = [dict(type='ConfusionMatrix'), ...]
```

**Reply by Xuchangsheng1** [reply]
Where is inserted into?

**Reply by Ezra-Yu** [reply]
in the dataset section.

**Reply by whdcumt** [reply]
> in config file:
> 
> ```
> val_evaluator = dict(type='ConfusionMatrix')
> ```
> 
> or
> 
> ```
> val_evaluator = [dict(type='ConfusionMatrix'), ...]
> ```
This method should be set in version 0.x，In version 1.x, this setting will be reported with an error，，In version 1.x，Under the folder /mmcls/evaluation/metrics, there are 'Accuracy', 'SingleLabelMetric', 'MultiLabelMetric', 'AveragePrecision',
    'MultiTasksMetric', 'VOCAveragePrecision', 'VOCMultiLabelMetric'
My question is, 'Accuracy', 'SingleLabelMetric',Whether they can be output at the same time,There is no problem with single metric output，Also, is there code for calculating AUC values and plotting VOC curves

**Reply by whdcumt** [reply]
I got it in the instruction manual

**Reply by Ezra-Yu** [reply]
there is a PR https://github.com/open-mmlab/mmpretrain/pull/1345/files#diff-3f842b63c42e45106f28b9b3c7069a1a739a747e98c332d158c4a802279f88b6 for AUC

**Comment by SaltAdamW** [comment]
you can redefine the metric and print it
